### PR TITLE
cmd, packaging: import BPF headers from kernel, detect whether host headers are usable

### PR DESCRIPTION
**Pull request by bboozzoo**
Extracted from #10575. The branch imports BPF headers (linux/bpf.h, linux/bpf_common.h) from the Linux kernel v5.13-11855-g77d34a4683b0 in an unchanged form. The bpf_insn.h is an import of samples/bpf/bpf_insn.h from the kernel source trree, further redacted to remove bits that are not strictly necessary for our use of BPF in device cgroup.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thanks! I like this and looks fine. I also like the idea of ./cmd/vendor, I am contemplating to use it for squashfuse in #10639 too

File: cmd/libsnap-confine-private/bpf-insn.h [hunk 1]
```diff
@@ -0,0 +1,311 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+/* imported from the Linux kernel, commit
+ * 77d34a4683b053108ecd466cc7c4193b45805528 (v5.13-11855-g77d34a4683b0) */
+
```
Comment (stolowski) [inline cmd/libsnap-confine-private/bpf-insn.h:5]: Maybe import it under a subdirectory, e.g. `libsnap-confine-private/linux/bpf-insn.h`?

**Review by stolowski — APPROVED** [review]
LGTM with one suggestion

File: cmd/libsnap-confine-private/bpf-insn.h (hunk 1 above)
Comment (bboozzoo) [reply cmd/libsnap-confine-private/bpf-insn.h:5]: Or `libsnap-confine-private/bpf/`? Also I could probably move the vendored headers to `libsnap-confine-private/bpf/vendor`, @mvo5 wdyt?

Comment (mvo5) [reply cmd/libsnap-confine-private/bpf-insn.h:5]: Yeah, sounds good. This location means that I will have to find my own place to vendor "squashfuse" but that is ok, I think the dir layout suggested here makes sense :)

Comment (stolowski) [reply cmd/libsnap-confine-private/bpf-insn.h:5]: > Or `libsnap-confine-private/bpf/`? Also I could probably move the vendored headers to `libsnap-confine-private/bpf/vendor`, @mvo5 wdyt?

Yes, either works, my main point was to have it separated from our headers.

**Comment by bboozzoo** [comment]
@mvo5 the failures are unrelated, please merge

**Pull request merged**